### PR TITLE
Fix HIX value diff when page content has nested tags

### PR DESCRIPTION
**Pull request by seluianova**
### Short description
Using root.iter() was not a good choice to iterate over the paragraphs, because it goes "one-way" deep into the nested elements.
For example, if a page has the following structure:
`<div><p>Some image</p><p><a href="some.image"></a></p><p>&nbsp;</p></div>`
The iteration will proceed as follows:
```
<div>
<p>
<p>
<a>
<img>
--> end of iteration
```
And the last p-node will not be processed.

### Proposed changes
Use list(root) instead of root.iter()

### Side effects
No?

### Resolved issues
Another follow-up to #2577

**UPD**: added another small HIX-related fix
Fixes: #2621

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Comment by seluianova** [comment]
Added another small HIX-related fix for #2621

File: integreat_cms/release_notes/current/unreleased/2621.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+en: Outdate HIX value when reversing changes in the editor
+de: Veralte HIX-Wert beim Rückgängigmachen von Änderungen im Editor
```
Comment (seluianova) [inline integreat_cms/release_notes/current/unreleased/2621.yml:2]: Not sure if it sounds okay in German?

Comment (timobrembeck) [reply integreat_cms/release_notes/current/unreleased/2621.yml:2]: Maybe like this?
```suggestion
de: Markiere HIX-Wert als veraltet, wenn Änderungen im Editor rückgängig gemacht werden
```

Also the English text could be made a bit more clear, but I would also be fine with the current wording.
```
en: Mark HIX value as outdated when changes are undone in the editor
```

**Review by timobrembeck — APPROVED** [review]
Thanks! :+1:

**Review by JoeyStk — APPROVED** [review]
Thank you very much. LGTM!

**Pull request merged**